Bottom: 1f32b4e14bbc4ade2f8a499989f4aedb2a039fb8
Top:    460b00dddc582a3b2243f9b4de746b49654146a7
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-05 15:55:49 +0100

Refresh of hid-gtk-replace-gtk_widget_rea

---

diff --git a/src/hid/gtk/gui-pinout-preview.c b/src/hid/gtk/gui-pinout-preview.c
index c4bef20..79fc0f2 100644
--- a/src/hid/gtk/gui-pinout-preview.c
+++ b/src/hid/gtk/gui-pinout-preview.c
@@ -191,14 +191,14 @@ ghid_pinout_preview_set_property (GObject * object, guint property_id,
 				  const GValue * value, GParamSpec * pspec)
 {
   GhidPinoutPreview *pinout = GHID_PINOUT_PREVIEW (object);
+  GdkWindow *window = gtk_widget_get_window (GTK_WIDGET (pinout));
 
   switch (property_id)
     {
     case PROP_ELEMENT_DATA:
       pinout_set_data (pinout, (ElementType *)g_value_get_pointer (value));
-      if (GTK_WIDGET_REALIZED (GTK_WIDGET (pinout)))
-        gdk_window_invalidate_rect (
-            gtk_widget_get_window (GTK_WIDGET (pinout)), NULL, FALSE);
+      if (window != NULL)
+        gdk_window_invalidate_rect (window, NULL, FALSE);
       break;
     default:
       G_OBJECT_WARN_INVALID_PROPERTY_ID (object, property_id, pspec);
